framework/base/Event.php

```diff
@@ -192,5 +203,17 @@ public static function trigger($class, $name, $event = null)
                 }
             }
         } while (($class = get_parent_class($class)) !== false);
```

Comment:
In original algorithm check for event listeners is performed in the following way:

1. Listeners for $class
2. Listeners for $class's parents
3. Listeners for $class's parents's parentents
4. So on...

If we will use `class_parents()` an order will be unpredictable